src/Features/Blockcore.Features.Wallet/WalletManager.cs

```diff
@@ -2087,5 +2097,102 @@ public ExtKey GetExtKey(WalletAccountReference accountReference, string password
 
             return new ExtKey(privateKey, wallet.ChainCode);
         }
+
+        /// <inheritdoc />
+        public IEnumerable<string> Sweep(IEnumerable<string> privateKeys, string destAddress, bool broadcast)
```

Comment:
Good job.

I wonder if the Sweep has to be part of the WalletManager?
It could get its own class for sweeping that asks the manager for next address.
This will pollute less the manager which is already massive.

Thoughts? (also for @zeptin  )

Reply:
Good question. I was wondering the same thing myself. StratX has a separate layer that we don't have called WalletService. Putting it in manager like I did effectively couples WalletManager with BlockStore which I'm not sure is bad or irrelevant. Not sure what kind of wallet you can have a wallet without a blockstore. Open to suggestions. If you think a separate dependency in the wallet feature is better I can go with that.